src/v/pandaproxy/reply.h

```diff
@@ -76,50 +77,79 @@ errored_body(std::error_condition ec, ss::sstring msg) {
     return rep;
 }
 
+inline std::unique_ptr<ss::httpd::reply>
+errored_body(std::error_condition ec, ss::sstring msg, error_probe& eprobe) {
+    auto rep = errored_body(ec, std::move(msg));
+
+    using status_type = ss::httpd::reply::status_type;
+
+    // See seastar/http/reply.hh::reply::status_type for a list
+    // of supported error status codes.
+    if (rep->_status >= status_type{500}) eprobe.increment_5xx();
+    else if (rep->_status >= status_type{400}) eprobe.increment_4xx();
+    else if (rep->_status >= status_type{300}) eprobe.increment_3xx();
+
+    return rep;
+}
+
 inline std::unique_ptr<ss::httpd::reply>
 errored_body(std::error_code ec, ss::sstring msg) {
     return errored_body(make_error_condition(ec), std::move(msg));
 }
 
+inline std::unique_ptr<ss::httpd::reply>
+errored_body(std::error_code ec, ss::sstring msg, error_probe& eprobe) {
+    return errored_body(make_error_condition(ec), std::move(msg), eprobe);
+}
+
 inline std::unique_ptr<ss::httpd::reply> unprocessable_entity(ss::sstring msg) {
     return errored_body(
       make_error_condition(reply_error_code::kafka_bad_request),
       std::move(msg));
 }
```

Comment:
This is unused

Reply:
I'm using a different approach.
Changes at https://github.com/redpanda-data/redpanda/pull/5422